src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -575,15 +577,197 @@ public String getName() {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th != null) {
+				th.interrupt();
+			}
+			return CVoid.VOID;
 		}
 
 		@Override
 		public Version since() {
 			return MSVersion.V3_3_4;
 		}
 
+		@Override
+		public String getName() {
+			return "x_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {mixed thread} Interrupts this thread. In other words \"interrupted status\" set true."
+					+ " The \"interrupted status\" can be checked using the method: " + x_is_interrupted.class.getSimpleName() + "()"
+					+ " If this thread is blocked in an invocation of the " + Scheduling.sleep.class.getSimpleName() + "(), " + x_thread_join.class.getSimpleName() + "() methods"
+					+ " then its \"interrupt status\" will be cleared and it will receive an InterruptedException.";
+		}
 	}
 
+	@api
+	@seealso({x_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		private static java.lang.reflect.Method interrupt;
+		static {
+			try {
+				interrupt = Thread.class.getDeclaredMethod("isInterrupted", boolean.class);
+				interrupt.setAccessible(true);
+			} catch (Exception e) {
+				//never happened
+				throw new CancelCommandException(e.getMessage(), Target.UNKNOWN);
+			}
+		}
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			Thread th;
+			boolean clearFlag = true;
+			if(args.length == 2) {
+				th = THREAD_ID_MAP.get(ArgumentValidation.getString(args[0], t));
+				clearFlag = ArgumentValidation.getBooleanish(args[1], t);
+			} else if(args.length == 1) {
+				if(args[0] instanceof CBoolean) {
+					th = Thread.currentThread();
+					clearFlag = ((CBoolean) args[0]).getBooleanValue(t);
+				} else {
+					th = THREAD_ID_MAP.get(ArgumentValidation.getString(args[0], t));
+				}
+			} else {
+				th = Thread.currentThread();
+			}
+			if(th != null) {
+				try {
+					return CBoolean.get((Boolean) interrupt.invoke(th, clearFlag));
+				} catch (IllegalAccessException | InvocationTargetException e) {
+					//never happened
+					throw new CancelCommandException(e.getMessage(), t);
+				}
+			} else {
+				return CBoolean.TRUE;
+			}
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_is_interrupted";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1, 2};
+		}
+
+		@Override
+		public String docs() {
+			return "bool {mixed thread=current, bool clearFlag=true} Tests whether this thread has been interrupted."
```

Comment:
Giving default values like this is convenient, but not consistent with other functions. Optional arguments are put between `[]`s, and the description tells what happens when they are omitted. There are examples of this in the API.